src/ERC6538Registry.sol

```diff
@@ -0,0 +1,45 @@
+// SPDX-License-Identifier: UNLICENSED
+pragma solidity 0.8.20;
+
+import {IERC6538Registry} from "./interfaces/IERC6538Registry.sol";
+
+/// @dev `ERC6538Registry` contract to map accounts to their stealth meta-address. See
+/// [ERC-6538](https://eips.ethereum.org/EIPS/eip-6538) to learn more.
+contract ERC6538Registry is IERC6538Registry {
+  /// @notice Maps a registrant's identifier to the scheme ID to the stealth meta-address.
+  /// @dev `registrant` may be a standard 160-bit address or any other identifier.
+  /// @dev `schemeId` is an integer identifier for the stealth address scheme.
+  mapping(bytes registrant => mapping(uint256 schemeId => bytes)) public stealthMetaAddressOf;
+
+  /// @inheritdoc IERC6538Registry
+  function registerKeys(uint256 schemeId, bytes memory stealthMetaAddress) external {
+    bytes memory registrant = _toBytes(msg.sender);
+    stealthMetaAddressOf[registrant][schemeId] = stealthMetaAddress;
```

Comment:
In ERC-5564 "A mapping from the schemeId to its specification MUST be declared in the EIP that proposes to standardize a new stealth address scheme. " Does this mean at some point we will need to fetch from a scheme registry contract to validate the passed in schemeId?

Also, can a user register an address to the wrong scheme? Would this cause any issues for downstream consumers of the event

Reply:
Or if the stealthMetaAddress is not formatted correctly would that cause issues

Reply:
Good questions

> Does this mean at some point we will need to fetch from a scheme registry contract to validate the passed in schemeId?

I don't think we need to validate the scheme ID passed in.

Doing so would add a lot of complexity: either an oracle to fetch off-chain data, or change the EIP so the scheme ID lives on-chain, but then the question is which chain and how do you read cross-chain data for arbitrary chains.

> Also, can a user register an address to the wrong scheme? Would this cause any issues for downstream consumers of the event

Yea there is nothing preventing registering the wrong or a non-existent scheme ID. There's not much we can do here, but you're right that it could result in inaccessible funds so we just have to document it.